app/src/main/java/com/onboarding/rick_and_morty_app/viewmodel/SplashViewModel.kt

```diff
@@ -0,0 +1,37 @@
+package com.onboarding.rick_and_morty_app.viewmodel
+
+import androidx.lifecycle.MutableLiveData
+import androidx.lifecycle.ViewModel
+import androidx.lifecycle.viewModelScope
+import com.onboarding.rick_and_morty_app.util.Event
+import kotlinx.coroutines.Dispatchers.IO
+import kotlinx.coroutines.launch
+import kotlinx.coroutines.withContext
+
+class SplashViewModel : ViewModel() {
+    private var mutableLiveDataSplash: MutableLiveData<Event<Data<Boolean>>> = MutableLiveData()
+
+    fun getLiveDataSplash() = mutableLiveDataSplash
+
+    fun loadMainActivity() = viewModelScope.launch {
+        withContext(IO) {
+            Thread.sleep(LOAD_TIME)
```

Comment:
you should use delay()... thread.sleep blocks the entire thread, while delay only the corutine

Reply:
replaced